kong/reports.lua

```diff
@@ -143,6 +157,12 @@ local function send_report(signal_type, t, host, port)
   if not ok then
     return
   end
+  if tls then
+    local opts = {
+      ssl_verify = false,
+    }
+    sock:tlshandshake(opts)
```

Comment:
Is there any reason that's preventing us from enabling TLS verification?

Also, if `tlshandshake` fails, we should probably bail out and not continue anymore.

Reply:
>  Also, if tlshandshake fails, we should probably bail out and not continue anymore.

see -> https://github.com/Kong/kong/pull/8159#discussion_r764821364

Reply:
> Is there any reason that's preventing us from enabling TLS verification?

I discussed this with @hutchic and we took the safe route as we can't be _always_ sure that certificates are up to date on the host that runs this code

Reply:
Should it bailout if handshake fails?

Reply:
+1 on @bungle , continuing after `sock:tlshandshake` fails seems to be a bad idea.

Reply:
Also, for the frequency of reports we send, does enabling TLS session reuse for this connection makes sense? If so it would be a simple change on the call here. Ping @hutchic